compose: Add feature to insert global time

yash-agarwa-l opened the pull request:
Fixes #1215.

<!-- Before vs After -->
<table>
  <caption><strong>Before vs After</strong></caption>
  <tbody>
    <tr>
      <td align="center">
        <strong>Before</strong><br />
        <img
          src="https://github.com/user-attachments/assets/537108b6-a88d-4fa9-99b3-b92ca07caa7a"
          width="300"
          alt="Before state"
        />
      </td>
      <td align="center">
        <strong>After</strong><br />
        <img
          src="https://github.com/user-attachments/assets/d4f996ca-36e2-431f-8291-62bea91b808c"
          width="300"
          alt="After state"
        />
      </td>
    </tr>
  </tbody>
</table>

<br />

<!-- After - Additional Screens -->
<table>
  <caption><strong>After – Additional</strong></caption>
  <tbody>
    <tr>
      <td align="center">
        <img
          src="https://github.com/user-attachments/assets/ddf5f464-a0f5-4d8b-90d9-b0a4346e34da"
          width="300"
          alt="After additional view 1"
        />
      </td>
      <td align="center">
        <img
          src="https://github.com/user-attachments/assets/7b1ff993-66ae-421f-adad-caf805d0828e"
          width="300"
          alt="After additional view 2"
        />
      </td>
    </tr>
    <tr>
      <td align="center">
        <img
          src="https://github.com/user-attachments/assets/fc065c78-6040-43c6-a87f-0384f574375c"
          width="300"
          alt="After additional view 3"
        />
      </td>
      <td align="center">
        <img
          src="https://github.com/user-attachments/assets/e85777e1-cef9-4186-b471-bbf7a702807a"
          width="300"
          alt="After additional view 4"
        />
      </td>
    </tr>
  </tbody>
</table>

<br />

<!-- Videos -->
<table>
  <tbody>
    <tr>
      <td align="center">
        <strong>24 Hours</strong><br />
        <video src="https://github.com/user-attachments/assets/0a508504-967a-464b-923c-2343bdaa81cb" width="400" controls></video>
      </td>
      <td align="center">
        <strong>12 Hours</strong><br />
        <video src="https://github.com/user-attachments/assets/f915f323-9836-44c3-b780-1c780ccee133" width="400" controls></video>
      </td>
    </tr>
  </tbody>
</table>

alya commented:
As usual, please include before/after _screenshots_ demonstrating all your changes, in addition to any videos.

yash-agarwa-l commented:
Thank you, I’ve added the after screenshot as well.

chrisbobbe commented on lib/widgets/compose_box.dart:
```diff
@@ -1620,13 +1644,73 @@ sealed class ComposeBoxController {
 }
 
+
+
```
nit: don't add extra lines

alya commented:
Are we controlling the date/time picker styling at all? The "Select date/time" text should perhaps be sized like a heading.

yash-agarwa-l commented:
Thanks for the review. I’ve addressed the feedback and squashed the commits as requested.

alya commented:
I dunno, you can try our modal heading size and update the screenshots for review.

yash-agarwa-l commented:
Thanks @alya. I’ve updated the PR description with the new screenshots.
Is that good? If so, I’ll update the implementation to match.

yash-agarwa-l commented:
The “Select date/time” text has been updated with a modal heading size.

Previously,
<img
  src="https://github.com/user-attachments/assets/1d460aad-5994-4228-90b7-6748bd2a8d8c"
  alt="Select date/time modal with heading styling"
  width="300"
/>

yash-agarwa-l commented:
Yes, by using the `twentyFourHourTime` parameter, I have updated it to use a different time picker format according to the user's settings. I have also added a video and a screenshot.
